### ISSUE TITLE
Shell script: # followed by non-whitespace may or may not be comment

### ISSUE THREAD

**Issue by J-Siu** [rioj7/remove-comments#24]
Version 1.13.1

When `Remove All Comments`, following line (from `p10k.zsh`):
  ```
  (( $#branch > 32 )) && branch[13,-13]="…"  # <-- this line
  ```
becomes:
  ```
  (( $
  ```

The correct outcome should be:
  ```
  (( $#branch > 32 )) && branch[13,-13]="…"
  ```

Additional information:
1. In shell script(posix), if `#` immediately followed by non-white-space, it may or may not not be comment.
2. Complexity of shell script:
    ```
    # Following line does not contain comment
    cat file | grep -v -e ^# -e ^$
    # Following line is partial comment
    cat file | grep -v -e ^# -e ^$ ### This part is a comment
    ```

**Comment by rioj7** [comment]
@J-Siu I have tried these strings in VSC and found that syntax highlighting correctly detects the comments and command arguments.

I have looked a the TextMate Grammar and it is not easy to filter these `#` characters.
It looks like it has transformed a grammar for bash to a collection of `begin`-`end` regex and specify which non-terminals are possible after this `begin`, if non matches any more it must find the `end`.

If I read the TextMate Grammar correct it also accepts `//` as single line comment.

**Comment by J-Siu** [comment]
@rioj7 Understand. I think unless you can leverage some existing tool like "shell syntax", "shell format", or some parsing library, it will be very complicated.

One no so ideal workaround is add a "Remove All Single Line Comments", which ignore in-line comments, and add full support of shell script in future.

**Comment by rioj7** [comment]
@J-Siu I can make the lines that have the first non whitespace a `#` the comment lines

**Comment by J-Siu** [comment]
@rioj7 I actually see there is a "Remove All Single Line Comments" already. So the workaround already exist. Not sure why I missed it before🤦‍♂️

**Comment by rioj7** [comment]
@J-Siu A `Single Line Comment` is a comment that starts and ends on the same line, all comments with `#` in bash are single line comments. I have to define the comments with a regex to get the required behavior.

**Comment by J-Siu** [comment]
@rioj7 Yes. "Remove All Single Line Comments" currently also remove in-line comments.

**Comment by rioj7** [comment]
@J-Siu in v1.13.2 I have set that only lines starting with `#` are removed

**Comment by J-Siu** [comment]
@rioj7 "Remove All Single Line Comments" is working correctly now.
